From: Duncan Eastoe <deastoe@vyatta.att-mail.com>
Date: Tue, 13 Nov 2018 15:26:21 +0000
Subject: zebra: Add table_id to rib_table_info_t

When given a route_table this allows the corresponding kernel table
ID to be determined. The table_id value is set upon table creation
to the table_id of the VRF, unless the table was created with a
specific ID.
---
 zebra/rib.h          | 1 +
 zebra/zebra_router.c | 1 +
 2 files changed, 2 insertions(+)

diff --git a/zebra/rib.h b/zebra/rib.h
index 927675e..19a8f69 100644
--- a/zebra/rib.h
+++ b/zebra/rib.h
@@ -284,6 +284,7 @@ typedef struct rib_table_info_t_ {
 	struct zebra_vrf *zvrf;
 	afi_t afi;
 	safi_t safi;
+	uint32_t table_id;
 
 } rib_table_info_t;
 
diff --git a/zebra/zebra_router.c b/zebra/zebra_router.c
index a891ffb..0a5f88b 100644
--- a/zebra/zebra_router.c
+++ b/zebra/zebra_router.c
@@ -117,6 +117,7 @@ struct route_table *zebra_router_get_table(struct zebra_vrf *zvrf,
 	info->zvrf = zvrf;
 	info->afi = afi;
 	info->safi = safi;
+	info->table_id = tableid;
 	route_table_set_info(zrt->table, info);
 	zrt->table->cleanup = zebra_rtable_node_cleanup;
 
